localstack-core/localstack/services/lambda_/event_source_mapping/esm_worker_factory.py

```diff
@@ -0,0 +1,148 @@
+from localstack.aws.api.lambda_ import (
+    EventSourceMappingConfiguration,
+)
+from localstack.aws.api.pipes import (
+    DynamoDBStreamStartPosition,
+    KinesisStreamStartPosition,
+    PipeSourceDynamoDBStreamParameters,
+    PipeSourceKinesisStreamParameters,
+    PipeSourceParameters,
+    PipeSourceSqsQueueParameters,
+    PipeTargetInvocationType,
+    PipeTargetLambdaFunctionParameters,
+    PipeTargetParameters,
+)
+from localstack.services.lambda_.event_source_mapping.esm_event_processor import (
+    EsmEventProcessor,
+)
+from localstack.services.lambda_.event_source_mapping.esm_worker import EsmStateReason, EsmWorker
+from localstack.services.lambda_.event_source_mapping.pipe_loggers.noops_pipe_logger import (
+    NoOpsPipeLogger,
+)
+from localstack.services.lambda_.event_source_mapping.pipe_utils import (
+    get_internal_client,
+    get_standardized_service_name,
+)
+from localstack.services.lambda_.event_source_mapping.pollers.dynamodb_poller import DynamoDBPoller
+from localstack.services.lambda_.event_source_mapping.pollers.kinesis_poller import KinesisPoller
+from localstack.services.lambda_.event_source_mapping.pollers.sqs_poller import SqsPoller
+from localstack.services.lambda_.event_source_mapping.senders.lambda_sender import LambdaSender
+from localstack.utils.aws.arns import parse_arn
+from localstack.utils.aws.client_types import ServicePrincipal
+
+
+class EsmWorkerFactory:
+    event_source_mapping_config: EventSourceMappingConfiguration
+    function_role_arn: str
+    enabled: bool
+
+    def __init__(self, event_source_mapping_config, function_role, enabled):
+        self.event_source_mapping_config = event_source_mapping_config
```

Comment:
Called `esm_config` in other classes, could we standardize this?

Reply:
makes sense 👍 (leftover from trying out both)

Reply:
renamed to `esm_config`